src/main/java/com/trolley/trolley/InvoiceGateway.java

```diff
@@ -0,0 +1,225 @@
+package com.trolley.trolley;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.trolley.Exceptions.InvalidFieldException;
+import com.trolley.trolley.Invoice.SearchBy;
+import com.trolley.trolley.types.Invoices;
+
+public class InvoiceGateway
+{
+    Client client;
+    
+    public InvoiceGateway(final Configuration config) {
+        this.client = new Client(config);
+    }
+    
+    /**
+     * Create a new invoice with data provided by the invoice object
+     * @param invoiceBody
+     * @return Invoice created invoice
+     * @throws Exception
+     */
+    public Invoice create(final Invoice invoiceBody) throws Exception {
+        String body = invoiceBody.toString();
+        
+        final String endPoint = "/v1/invoices/create/";
+        final String response = this.client.post(endPoint, body);
+        return Invoice.invoiceFactory(response);
+    }
+    
+    /**
+     * Update the invoice represented by the given invoiceId with data found in 
+     * invoiceBody object supplied.
+     * Only the fields that you send will be updated.
+     * @param invoiceId
+     * @param invoiceBody
+     * @return Invoice updated invoice
+     * @throws Exception
+     */
+    public Invoice update(final String invoiceId, final Invoice invoiceBody) throws Exception {
+        invoiceBody.setInvoiceId(invoiceId);
+        String body = invoiceBody.toString();
+
+        final String endPoint = "/v1/invoices/update/";
+        final String response = this.client.post(endPoint, body);
+
+        return Invoice.invoiceFactory(response);
+    }
+    
+    /**
+     * Delete one invoice by its ID
+     * @param invoiceId
+     * @return boolean true if the delete operation was successful
+     * @throws Exception
+     */
+    public boolean delete(final String invoiceId) throws Exception {
+        final String endPoint = "/v1/invoices/delete/";
+        
+        ArrayList<String> invoiceIds = new ArrayList<String>() {
```

Comment:
I would suggest having separate classes for request objects, something like this: 
`public class DeleteInvoicesRequest {
public DeleteInvoicesRequest(List<String> value) {
invoiceIds = value
}
public List<String> invoiceIds
}

DeleteInvoicesRequest requestBody = new DeleteInvoicesRequest(invoiceId)
String body = new ObjectMapper().writeValueAsString(requestBody)
`
That will make code much more readable and easily maintainable.

Reply:
Agreed, but I plan to introduce this change SDK-wide in the next MAJOR release since it will be a breaking change.
This release is supposed to be covering Invoice API using the existing usage pattern